### PR TITLE
Revised Inner Outer Inner mode for Classic mode

### PR DESCRIPTION
**Pull request by igiannakas**
Ported the Arachne enhancements to inner-outer-inner wall ordering mode to Classic. It should behave better when multiple external perimeters are present in the same island and gracefully fall back to outer first to preserve wall consistency when inner-outer-inner wall ordering cannot be respected.

This brings feature parity between Arachne and Classic wall ordering for inner outer inner.

### CODE REVIEW

**Comment by igiannakas** [comment]
Test done, benchy looks good!

**Comment by Eldenroot** [comment]
Nice! thx!

**Comment by SoftFever** [comment]
@igiannakas 
Do you have screenshot to demonstrate the improvement for the scenario you mentioned?

**Comment by igiannakas** [comment]
Yeap:
**Before:**
Middle perimeter drawn before outer
image

**After:**
Outer perimeter drawn correctly before neighbouring inner perimeter
image

**Before:**
Middle perimeter drawn before outer
image

**After:**
Outer perimeter drawn correctly before neighbouring inner perimeter
image

**Comment by igiannakas** [comment]
The above is the same issue that Arachne used to have before the revisions in the inner/outer/inner ordering mode -> basically middle perimeters are on occasion drawn before the outer perimeter when there are multiple external perimeters on the same island. 

The revision enforces middle perimeters to be drawn after the external and if that is not possible it falls back to drawing the external perimeter first to minimise surface artefacts caused by the middle perimeter pushing out on the external ones if drawn first.

Basically porting the below to classic mode and the fixes that came from those:
https://github.com/SoftFever/OrcaSlicer/pull/2276
https://github.com/SoftFever/OrcaSlicer/pull/1861
https://github.com/SoftFever/OrcaSlicer/pull/1352

**Review by SoftFever — APPROVED** [review]
Looks good.
Thank you
